soc/xtensa/intel_adsp/Kconfig

```diff
@@ -133,4 +133,12 @@ config ADSP_IDLE_CLOCK_GATING
 	  (wait for interrupt) during idle, the clock can be gated (however, this
 	  does not mean that this will happen).
 
+config ADSP_DYNAMIC_CLOCK_SWITCHING
+	bool "DSP clock switching in Idle"
+	depends on PM_STATE_CUSTOM_DATA
+	help
+	  When true, FW will switch clock if all cores are in idle state. The dynamic
```

Comment:
@marc-hb Please take a look at the updated version to see if it's what you had in mind?

Reply:
That's a perfect Kconfig help thank you. I wish they were all like this :-)

Now I wonder why it's not enabled by default but I guess it's either because it's a bit early still or because I missed something else so nevermind.

In case you were wondering, `default y` does NOT override `depends on PM_STATE_CUSTOM_DATA`.